spec/form_objects/forms/payment_link_form_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "rails_helper"
+
+RSpec.describe Forms::PaymentLinkForm, type: :model do
+  describe "Payment URL" do
+    context "when form is live" do
+      let(:form) do
+        build(:form, :live)
+      end
+
+      it "validates the URL" do
+        payment_link_form = described_class.new(form:, payment_url: "gov.uk")
+        error_message = I18n.t("errors.messages.url")
+
+        payment_link_form.validate(:payment_url)
+
+        expect(payment_link_form.errors.full_messages_for(:payment_url)).to include(
+          "Payment url #{error_message}",
+        )
+      end
+
+      it "is valid if blank" do
+        payment_link_form = described_class.new(form:, payment_url: "")
+
+        expect(payment_link_form).to be_valid
+      end
+
+      it "is valid if a PAY url is given" do
+        payment_link_form = described_class.new(form:, payment_url: "https://www.gov.uk/payments/test-org/test-service")
+
+        expect(payment_link_form).to be_valid
+      end
```

Comment:
We need to include examples that are _not_ valid otherwise we don't know if our validation rule is correct.